forgerock-openbanking-uk-aspsp-as/forgerock-openbanking-uk-aspsp-as-gateway/forgerock-openbanking-uk-aspsp-as-gateway-server/src/main/java/com/forgerock/openbanking/aspsp/as/service/registrationrequest/RegistrationRequestJWTClaims.java

```diff
@@ -0,0 +1,211 @@
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.forgerock.openbanking.aspsp.as.service.registrationrequest;
+
+import com.forgerock.openbanking.common.error.exception.dynamicclientregistration.DynamicClientRegistrationErrorType;
+import com.forgerock.openbanking.common.error.exception.dynamicclientregistration.DynamicClientRegistrationException;
+import com.nimbusds.jwt.JWTClaimsSet;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.util.StringUtils;
+
+import java.text.ParseException;
+import java.util.Date;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import static com.forgerock.openbanking.constants.OpenBankingConstants.SSAClaims;
+
+@Slf4j
+public class RegistrationRequestJWTClaims {
+
+    private final JWTClaimsSet claimsSet;
+    private final JWTClaimsOrigin claimsOrigin;
+    private final DynamicClientRegistrationErrorType errorType;
+
+    public RegistrationRequestJWTClaims(JWTClaimsSet claimsSet, JWTClaimsOrigin claimsOrigin){
+        this.claimsSet = claimsSet;
+        this.claimsOrigin = claimsOrigin;
+        if(claimsOrigin == JWTClaimsOrigin.REGISTRATION_REQUEST_JWT){
+            this.errorType = DynamicClientRegistrationErrorType.INVALID_CLIENT_METADATA;
+        } else {
+            this.errorType = DynamicClientRegistrationErrorType.INVALID_SOFTWARE_STATEMENT;
+        }
+    }
+
+
+    /**
+     * getRequiredStringClaim Obtains a string containing the value for a specific claim. Catches the
+     * ParseExceptions thrown by nimbusds and instead throw an DynamicClientRegistrationException.
+     *
+     * @param claimName        @String the name of the SSA claim to obtain from the ssaClaims
+     * @return A non null or empty @String containing the ssa claim value
+     * @throws DynamicClientRegistrationException when the claim specified by the claimName does not exist, or is empty.
+     */
+    public String getRequiredStringClaim(String claimName) throws DynamicClientRegistrationException {
+        Optional<String> claimValue = getOptionalStringClaim(claimName);
+        if(!stringClaimValueIsValid(claimValue)){
+            String errorString = "Claim '" + claimName + "' was not set in " + claimsOrigin + ". This claim " +
+                    "is required";
+            log.debug("getStringClaims() {}", errorString);
```

Comment:
Could be better log.error or log.warn instead of log.debug? because contains the error message.